src/VisualStudio/LiveShare/Impl/AbstractGoToDefinitionWithFarHandler.cs

```diff
@@ -18,19 +18,15 @@
 namespace Microsoft.VisualStudio.LanguageServices.LiveShare
 {
     /// <summary>
-    /// Handler for a goto defintion request. This can return files outside the shared
-    /// folder cone. The HostProtocolConverter handles converting files outside the shared
-    /// cone to the vslsexternal scheme.
-    /// This lives in external access because it has a dependency on FAR, which requires the UI thread.
+    /// Typescript uses FAR for goto def as they do not implement the definition service.
+    /// So implement goto def for typescript using FAR.
+    /// FAR has a UI thread dependency.
     /// </summary>
-    internal class GotoDefinitionHandler : ILspRequestHandler<LSP.TextDocumentPositionParams, object, Solution>
+    internal abstract class AbstractGoToDefinitionWithFarHandler : ILspRequestHandler<LSP.TextDocumentPositionParams, object, Solution>
```

Comment:
We don't abbreviate in Roslyn.  So `Far` isn't something appropriate in a name.   from other discussions, it sounds like you're sitting on the FindUsagesService, so perhaps use that name here instead.